src/js/utils.js

```diff
@@ -16,6 +16,27 @@ exports.is_sample_genetically_altered = function is_sample_genetically_altered(d
   || datum.protein !== undefined;
 };
 
+exports.makeD3SVGElement = function(tag) {
+  return d3.select(document.createElementNS('http://www.w3.org/2000/svg', tag));
+};
+
+exports.stableSort = function(arr, cmp) {
```

Comment:
That's very interesting. Like you said though, it's convoluted, and you need to write a new one for each comparator-based sort you need to convert into a metric-based sort.